libraries/AP_Relay/AP_Relay_Params.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <AP_Param/AP_Param.h>
+#include "AP_Relay_config.h"
+
+class AP_Relay_Params {
+public:
+    static const struct AP_Param::GroupInfo var_info[];
+
+    AP_Relay_Params(void);
+
+    /* Do not allow copies */
+    CLASS_NO_COPY(AP_Relay_Params);
+
+    enum class Default_State : uint8_t {
+        Off = 0,
+        On = 1,
+        NoChange = 2,
+    };
+
+    enum class Function : uint8_t {
+        none     = 0,
+        relay1   = 1,
+        relay2   = 2,
+        relay3   = 3,
+        relay4   = 4,
+        relay5   = 5,
+        relay6   = 6,
+        ignition = 7, // high for on
+        starter  = 8, // high for on
+        num_functions // must be the last entry
+    };
+
+    AP_Enum<Function> function;           // relay function
```

Comment:
AP_Enum is a int8, we ran out of servo functions and had to move to int16. We do have `AP_Enum16`. It will take a while for us to get up to that many, but if/when we do its annoying to fix.

Reply:
Sure.